controllers/recipe.js

```diff
@@ -0,0 +1,131 @@
+const { BadRequestError, NotFoundError } = require('../errors/index.js')
+const ObjectId = require('mongodb').ObjectId
+const Recipe = require('../models/Recipe.js')
+
+const getUserRecipes = async (req, res) => {
+    const { userId } = req.body
+    try {
+        const userRecipes = await Recipe.find({createdBy: userId})
+        res.json({userRecipes})
+    } catch (error) {
+        res.status(500).json({ message: 'Internal Server Error.' });
+    }
+}
+
+const getAllRecipes = async (req, res) => {
+    try {
+        const cuisineType = req.query.name
+        const allRecipes = await Recipe.find()
+        if (cuisineType) {
+            const filteredRecipes = allRecipes.filter(recipe => recipe.cuisineType + '=' === cuisineType);
+            res.json({filteredRecipes})
+        } else {
+            res.json({allRecipes})
+        }
+    } catch (error) {
+        res.status(500).json({ message: 'Internal Server Error.' });
+    }
+}
+
+const getRecipe = async (req, res) => {
+    const id = req.params.id
+    try {
+        const recipe = await Recipe.findOne({
+            _id: id,
+            createdBy: req.user.userId
+        });
+        if(!recipe) {
+            throw new NotFoundError('Recipe not found')
+        }
+        res.status(200).json({ recipe });
+    } catch (error) {
+        res.status(500).json({ message: 'Internal Server Error.' });
```

Comment:
You should keep the verification if the error is NotFoundError and return the 404 error, or return the error 500 when it isn't.
```
if (error instanceof NotFoundError) {
            res.status(404).json({ message: error.message });
        } else {
            console.error(error);
            res.status(500).json({ message: 'Internal Server Error.' });
        }
```